### PR TITLE
Automated build and release

### PR DESCRIPTION
**Pull request by nbrownus**
Whenever a new tag match `v[0-9]+.[0-9]+.[0-9]*` is pushed a new build will start and the resulting Android and iOS artifacts will be published to internal testing tracks. A draft release will also be created in github, we should publish the release if we promote either build to production.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,152 @@
+name: Create release and upload to Apple and Google
+on:
+  push:
+    tags:
+      # Only builds for tags with a meaningless build number suffix: v1.0.0-1
+      - 'v[0-9]+.[0-9]+.[0-9]*-.*'
```
Comment (JohnMaguire) [inline .github/workflows/release.yml:6]: I think the final two `*` should be `+` still. At minimum the first one.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+          TOKEN: ${{ secrets.MACHINE_USER_PAT }}
+        run: git config --global url."https://defined-machine:${TOKEN}@github.com".insteadOf "https://github.com"
+
+      - name: Install the appstore connect key material
+        env:
+          AC_API_KEY_SECRET_BASE64: ${{ secrets.AC_API_KEY_SECRET_BASE64 }}
+        run: |
+          AC_API_KEY_SECRET_PATH="$RUNNER_TEMP/key.p8"
+          echo "APP_STORE_CONNECT_API_KEY_KEY_FILEPATH=$AC_API_KEY_SECRET_PATH" >> $GITHUB_ENV
+          echo -n "$AC_API_KEY_SECRET_BASE64" | base64 --decode --output "$AC_API_KEY_SECRET_PATH"
```
Comment (JohnMaguire) [inline .github/workflows/release.yml:51]: Does this work? GNU base64 doesn't support `--output` AFAIK

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+      - name: Publish to Android internal track
+        run: |
+          cd android
+          fastlane release
+
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (brad-defined) [inline .github/workflows/release.yml:126]: I don't see this secret in the repo

**Review by JohnMaguire — DISMISSED** [review]
Can you do it again, but without pulling in Ruby?

File: .github/workflows/release.yml (hunk 3 above)
Comment (nbrownus) [reply .github/workflows/release.yml:126]: This one is provided by github on every run

File: .github/workflows/release.yml (hunk 2 above)
Comment (nbrownus) [reply .github/workflows/release.yml:51]: Works on a mac :D

**Review by brad-defined — APPROVED** [review]

**Pull request merged**